src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1931,6 +1908,8 @@ private long releaseAll() {
                 v.phase = sp;
                 if (v.parking != 0)
                     U.unpark(v.owner);
+                if (!all)
+                    break;
             }
```

Comment:
Stylistically, it might be cleaner to do this:

```suggestion
        do {
            WorkQueue[] qs; WorkQueue v; int sp, i;
            if ((sp = (int)c) == 0 || (qs = queues) == null ||
                qs.length <= (i = sp & SMASK) || (v = qs[i]) == null)
                break;
            if (c == (c = compareAndExchangeCtl(
                          c, ((UMASK & (c + RC_UNIT)) | (c & TC_MASK) |
                              (v.stackPred & LMASK))))) {
                v.phase = sp;
                if (v.parking != 0)
                    U.unpark(v.owner);
            } while(all);
```

Reply:
Yeah, except not quite that way, since the break should occur only if the CAS succeeds. I'll try other options.